pkg/convert/pot.go

```diff
@@ -110,47 +74,209 @@ func FromPot(b []byte) (model.Messages, error) {
 }
 
 func writeToPoTag(b *bytes.Buffer, tag poTag, str string) error {
-	encodedStr, err := json.Marshal(str) // use JSON encoding to escape special characters
-	if err != nil {
-		return fmt.Errorf("marshal %s: %w", tag, err)
+	switch tag { //nolint:exhaustive
+	case MsgStrPlural:
+		if err := writePluralForms(b, tag, str); err != nil {
+			return fmt.Errorf("write plural forms: %w", err)
+		}
+	default:
+		if err := writeDefault(b, tag, str); err != nil {
+			return fmt.Errorf("write default: %w", err)
+		}
 	}
 
-	encodedStr = encodedStr[1 : len(encodedStr)-1] // trim quotes
-	lines := strings.Split(string(encodedStr), "\\n")
+	return nil
+}
 
-	// Remove the empty string element. The line is empty when splitting by "\\n" and "\n" is the last character in str.
-	if lines[len(lines)-1] == "" {
-		lines = lines[:len(lines)-1]
-		lines[len(lines)-1] = lines[len(lines)-1] + "\\n" // add the "\n" back to the last line
+func writeDefault(b *bytes.Buffer, tag poTag, str string) error {
+	err, lines := getPoTagLines(str, tag)
+	if err != nil {
+		return fmt.Errorf("get po tag lines: %w", err)
 	}
 
 	if len(lines) == 1 {
-		_, err = fmt.Fprintf(b, "%s \"%s\"\n", tag, lines[0])
-		if err != nil {
+		if _, err = fmt.Fprintf(b, "%s \"%s\"\n", tag, lines[0]); err != nil {
 			return fmt.Errorf("write %s: %w", tag, err)
 		}
```

Comment:
Maybe use switch here to avoid nesting ?

```go
switch len(lines){
case 1:
...
default:
...}
```

Reply:
In this case, the if-else statement is more readable since there are only two branches.

Reply:
I just dont like these nested if
```go
if ... {
  if ...{
  }
}
```

This adds variety to it ?
```go
switch ...{
  case ...:
    if...{
  }
}
```
e.g.:
```go
    switch len(lines) {
    case 1:
        if _, err = fmt.Fprintf(b, "%s \"%s\"\n", tag, lines[0]); err != nil {
            return fmt.Errorf("write %s: %w", tag, err)
        }
    default:
        if _, err = fmt.Fprintf(b, "%s \"\"\n", tag); err != nil {
            return fmt.Errorf("write %s: %w", tag, err)
        }
        if err = writeMultiline(b, tag, lines); err != nil {
            return fmt.Errorf("write multiline: %w", err)
        }
    }
```

But that is not so important :)